src/Mod/Draft/draftguitools/gui_move.py

```diff
@@ -202,15 +202,39 @@ def move(self):
 
     def move_subelements(self):
         """Move the subelements."""
-        try:
-            if self.ui.isCopy.isChecked():
-                self.commit(translate("draft", "Copy"),
-                            self.build_copy_subelements_command())
-            else:
-                self.commit(translate("draft", "Move"),
-                            self.build_move_subelements_command())
-        except Exception:
-            _err(translate("draft", "Some subelements could not be moved."))
+        import Draft
+        #try:
+        if self.ui.isCopy.isChecked():
+            self.commit(translate("draft", "Copy"),
+                        self.build_copy_subelements_command())
+        else:
+            #self.commit(translate("draft", "Move"), # Moult implementation start
+            #            self.build_move_subelements_command()) # Moult implementation end
+
+            objects={}
+            
+            # create a dictionary with {'obj.Name' : 'list of selected SubObjects'}
+            for sel in self.selected_subelements:
+                if not sel.Object.Name in objects:
+                    objects[sel.Object.Name]=[]
+                if sel.SubObjects:
+                    objects[sel.Object.Name].extend(sel.SubElementNames)
+
+            App.ActiveDocument.openTransaction(translate("Draft","Move subelements"))
+
+            for name in objects: # for each object
+                # get the object and its shape
+                Gui.addModule("Draft")
+                Gui.doCommand('Draft.moveSubElements(' +
+                                'FreeCAD.ActiveDocument.' + name + ', ' +
+                                str(objects[name]) + ', ' +
+                                DraftVecUtils.toString(self.vector) +
+                                ')')
```

Comment:
This could be implemented with the `self.commit` mechanism as well.
```py
for name in objects:
    _cmd = 'Draft.move_sub_elements'
    _cmd += '('
    _cmd += 'FreeCAD.ActiveDocument.' + name + ', '
    _cmd += str(objects[name]) + ', '
    _cmd += DraftVecUtils.toString(self.vector)
    # _cmd += 'fix_invalid=' + str(fix_invalid)  # here you can implement another checkbox with the fix invalid option as described above
    _cmd += ')'
    self.commit(translate("Draft", "Move subelements"), _cmd)
```

And then you can remove `openTransaction` and `commitTransaction`.

Reply:
ok!